VAMobile/src/store/slices/authSlice.ts

```diff
@@ -94,6 +98,8 @@ export const initialAuthState: AuthState = {
   displayBiometricsPreferenceScreen: false,
   showLaoGate: false,
   authParamsLoadingState: AuthParamsLoadingStateTypeConstants.INIT,
+  requestNotificationsPreferenceScreen: false,
```

Comment:
I might be missing something, but it seems like `requestNotificationsPreferenceScreen` might be overlapping with `NOTIFICATION_COMPLETED_KEY` in async storage. Couldn't we just use `NOTIFICATION_COMPLETED_KEY` to determine whether to show the RequestNotificationScreen?

Reply:
Yeah, we could just use the value from async storage since it stays in sync with `requestNotificationsPreferenceScreen` anyway. To kill 2 birds with 1 stone, we could create a query for getting the `NOTIFICATION_COMPLETED_KEY` value, that way it'll be cached and easier to access. And when the app launches, it'll automatically load from async storage. Kinda like how we do with [useSystemNotificationsSettings](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/de30b5a69ae38c4a3dc3e68f3df98e46448e3e07/VAMobile/src/api/notifications/getSystemNotificationSettings.tsx#L11).